File: Data Structures & Algorithms/two-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let arr = [];

        for(let r = 0; r < nums.length; r++)
        {
            for(let c = r + 1; c < nums.length; c++)
            {
                if(nums[r] + nums[c] === target){
                    arr[0] = r;
                    arr[1] = c;
            }
            }
        }
        return arr;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def dfs(i, subset):
            if sum(subset) == target:
                res.append(subset.copy())
                return
            elif sum(subset) > target:
                return
            elif i == len(nums):
                return
            
            subset.append(nums[i])
            dfs(i, subset)
            subset.remove(nums[i])
            dfs(i + 1, subset)
        
        dfs(0, [])
        return res